### PR TITLE
Feature/i11054 clean submodules

### LINKED ISSUE #11054: Add option to clean submodules
### Feature description

It would be very handy if you could clean (not reset) all submodules (recursively) from git extensions. It's just one more tick box on the Commands->Clean working directory (or two if you want to control whether it is recursive or not). I think the relevant script is `git submodule foreach --recursive git clean -xfd`

### Environment

All
clean

### PR DESCRIPTION
**Pull request by NatRavenhill**
Fixes #11054 

## Proposed changes

-  Add "Clean submodules recursively" option
- When checked, the command "git submodule foreach --recursive git clean" will be ran after "git clean", with the same arguments

### Before

image

### After

image

## Test methodology 

- Tested with a couple of local projects
- Added a unit test in the GitCommands unit tests

## Test environment(s) 

- GIT 2.41.0.windows.1
- Windows 10

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/CommandsDialogs/FormCleanupRepository.Designer.cs [hunk 1]
```diff
@@ -28,231 +28,238 @@ protected override void Dispose(bool disposing)
         /// </summary>
         private void InitializeComponent()
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormCleanupRepository.Designer.cs:29]: Looks like you will have to fight with @scottsumrall for who gets the PR merged first (https://github.com/gitextensions/gitextensions/pull/11102), and who has to rebase. The rebase of the designer file is never pretty.

**Review by RussKie — APPROVED** [review]
👍

File: GitUI/CommandsDialogs/FormCleanupRepository.Designer.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/CommandsDialogs/FormCleanupRepository.Designer.cs:29]: Should be easier to merge this first, but the non essential changes should be reverted.

**Review by gerhardol — COMMENTED** [review]
preview is not handled in `Preview_Click()`
I suggest that is added too

File: GitUI/CommandsDialogs/FormCleanupRepository.cs [hunk 2]
```diff
@@ -43,18 +43,37 @@ public void SetPathArgument(string? path)
 
         private void Preview_Click(object sender, EventArgs e)
         {
-            var cleanUpCmd = GitCommandHelpers.CleanCmd(GetCleanMode(), dryRun: true, directories: RemoveDirectories.Checked, paths: GetPathArgumentFromGui());
+            string pathArgument = GetPathArgumentFromGui();
+            CleanMode mode = GetCleanMode();
+            var cleanUpCmd = GitCommandHelpers.CleanCmd(mode, dryRun: true, directories: RemoveDirectories.Checked, paths: pathArgument);
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormCleanupRepository.cs:48]: Please use the explicit type (applies below and to  Cleanup_Click() too).
This PR just add one call in each method, but the type should be clarified.
@mstv and @RussKie should write it down, I often forget myself.

```suggestion
            ArgumentString cleanUpCmd = GitCommandHelpers.CleanCmd(mode, dryRun: true, directories: RemoveDirectories.Checked, paths: pathArgument);
```

**Review by gerhardol — COMMENTED** [review]
@NatRavenhill and @scottsumrall can you agree how to merge the Designer changes?

**Comment by scottsumrall** [comment]
> @NatRavenhill and @scottsumrall can you agree how to merge the Designer changes?

I'm fairly new to this but happy to give it a shot. Would it possibly be easier to just revert my designer and modify based on what @NatRavenhill has? The UI wasn't particularly difficult to implement in the first place. Seems like merging the designer files could be problematic but it's not something I've really messed with before.

**Comment by gerhardol** [comment]
> > @NatRavenhill and @scottsumrall can you agree how to merge the Designer changes?
> 
> I'm fairly new to this but happy to give it a shot. Would it possibly be easier to just revert my designer and modify based on what @NatRavenhill has? The UI wasn't particularly difficult to implement in the first place. Seems like merging the designer files could be problematic but it's not something I've really messed with before.

The PR that goes second need to rebase on master using the Designer file from master and then manually reapply changes. Not so hard but often feeling like a waste of time. But hard to avoid here.
But if @scottsumrall is OK with it, we can merge this PR first.

**Comment by scottsumrall** [comment]
> > > @NatRavenhill and @scottsumrall can you agree how to merge the Designer changes?
> 
> > 
> 
> > I'm fairly new to this but happy to give it a shot. Would it possibly be easier to just revert my designer and modify based on what @NatRavenhill has? The UI wasn't particularly difficult to implement in the first place. Seems like merging the designer files could be problematic but it's not something I've really messed with before.
> 
> 
> 
> The PR that goes second need to rebase on master using the Designer file from master and then manually reapply changes. Not so hard but often feeling like a waste of time. But hard to avoid here.
> 
> But if @scottsumrall is OK with it, we can merge this PR first.

Works for me

**Comment by gerhardol** [comment]
Pushed a change for ArgumentString as well as for using common code for preview/execute.
Plan to merge tomorrow so #11102 can be rebased and merged.

File: GitUI/CommandsDialogs/FormCleanupRepository.cs (hunk 2 above)
Comment (gerhardol) [reply GitUI/CommandsDialogs/FormCleanupRepository.cs:48]: pushed a change

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you all

**Pull request merged**